### PR TITLE
Adds SHOULD_CALL_PARENT to Destroy()

### PR DESCRIPTION
**Pull request by ninjanomnom**
Timers, signals, and components need to know when the thing they're attached to is getting destroyed or bad things happen. Since any datum can make use of these every destroy needs to call parent.

The first commit is intentionally failing as it only adds the requirement. The second commit fixes the issues.

### CODE REVIEW

File: code/controllers/globals.dm [hunk 1]
```diff
@@ -20,9 +20,10 @@ GLOBAL_REAL(GLOB, /datum/controller/global_vars)
 
 	Initialize()
 
-/datum/controller/global_vars/Destroy()
-	//fuck off kevinz
```
Comment (Qustinnus) [inline code/controllers/globals.dm]: lold

File: code/controllers/globals.dm [hunk 2]
```diff
@@ -20,9 +20,10 @@ GLOBAL_REAL(GLOB, /datum/controller/global_vars)
 
 	Initialize()
 
-/datum/controller/global_vars/Destroy()
-	//fuck off kevinz
-	return QDEL_HINT_IWILLGC
```
Comment (ninjanomnom) [inline code/controllers/globals.dm]: This exploit needs to be fixed in another way which I'll look into

File: code/controllers/globals.dm [hunk 3]
```diff
@@ -20,9 +20,10 @@ GLOBAL_REAL(GLOB, /datum/controller/global_vars)
 
 	Initialize()
 
-/datum/controller/global_vars/Destroy()
-	//fuck off kevinz
-	return QDEL_HINT_IWILLGC
+/datum/controller/global_vars/Destroy(force)
+	if(!force)
+		return QDEL_HINT_IWILLGC
+	return ..()
```
Comment (SpaceManiac) [inline code/controllers/globals.dm]: #36287 explicitly made this impossible to delete, it should probably stay that way

**Comment by optimumtact** [comment]
You need to be able to supress this lint somehow for a given destroy call. @SpaceManiac is there anyway to supress a lint warning for one proc?

**Comment by ninjanomnom** [comment]
You can yes. `SHOULD_CALL_PARENT(0)`

**Comment by optimumtact** [comment]
just do that then

File: code/datums/weakrefs.dm [hunk 4]
```diff
@@ -16,8 +16,10 @@
 /datum/weakref/New(datum/thing)
 	reference = REF(thing)
 
-/datum/weakref/Destroy()
-	return QDEL_HINT_LETMELIVE	//Let BYOND autoGC thiswhen nothing is using it anymore.
+/datum/weakref/Destroy(force)
+	if(!force)
+		return QDEL_HINT_LETMELIVE	//Let BYOND autoGC thiswhen nothing is using it anymore.
+	return ..()
```
Comment (SpaceManiac) [inline code/datums/weakrefs.dm]: Should probably clean up the `weakref` var which points to it or else it will harddel

**Comment by ninjanomnom** [comment]
I don't like how we're preventing an admin abuse exploit by modifying unrelated core functionality but I guess that's more appropriate for another pr

**Review by SpaceManiac — APPROVED** [review]

**Pull request merged**